### PR TITLE
[browser][MT] Marshal resolved/unresolved tasks separately

### PR DESCRIPTION
**Pull request by pavelsavara**
This change will avoid sending 2 messages to target thread (UI) and instead will only send one.
- when the target thread is the same as current thread and this is synchronous/blocking return from JSExport
- when this is non-return parameter of JSImport <- this is new here

Otherwise will use new `JSAsyncTaskScheduler` 
- to make sure that the `mono_wasm_resolve_or_reject_promise_post` happens after `mono_wasm_invoke_jsimport_async_post`
- I'm not sure why [#99317](https://github.com/dotnet/runtime/pull/99317) was not enough and didn't help it.

contributes to https://github.com/dotnet/runtime/issues/98406

### CODE REVIEW

File: src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/Marshaling/JSMarshalerArgument.Task.cs [hunk 1]
```diff
@@ -140,13 +140,21 @@ internal void ToJSDynamic(Task? value)
         {
             Task? task = value;
 
+            var ctx = ToJSContext;
+
+            var isCurrentThreadOrParameter = ctx.IsCurrentThread() || slot.Type != MarshalerType.TaskPreCreated;
```
Comment (ilonatommy) [inline src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/Marshaling/JSMarshalerArgument.Task.cs:145]: Can we have a comment describing the parameter, like in the PR description? For me "non-return, JSImport" part is not clear without it.
```suggestion
            // current thread or non-return parameter of JSImport
            var isCurrentThreadOrParameter = ctx.IsCurrentThread() || slot.Type != MarshalerType.TaskPreCreated;
```

Comment (pavelsavara) [reply src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/Marshaling/JSMarshalerArgument.Task.cs]: I refactored it into `CanMarshalTaskResultOnSameCall` with detailed explanation

**Review by ilonatommy — APPROVED** [review]

**Comment by pavelsavara** [comment]
Filled https://github.com/dotnet/runtime/issues/99395

**Comment by pavelsavara** [comment]
CI issue is https://github.com/dotnet/runtime/issues/98721

**Pull request merged**